pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -429,19 +415,22 @@ func (r *ReconcileClusterPool) Reconcile(ctx context.Context, request reconcile.
 	return reconcile.Result{}, nil
 }
 
-// reconcileRunningClusters ensures the oldest pool.spec.runningCount unassigned clusters are
-// set to running, and the remainder are set to hibernating.
+// reconcileRunningClusters ensures the oldest unassigned clusters are set to running, and the
+// remainder are set to hibernating. The number of clusters we set to running is determined by
+// adding the cluster's configured runningCount to the number of unsatisfied claims for which we're
+// spinning up new clusters.
 func (r *ReconcileClusterPool) reconcileRunningClusters(
 	clp *hivev1.ClusterPool,
 	cds *cdCollection,
-	excessCount int,
+	extraRunning int,
 	logger log.FieldLogger,
 ) error {
 	// If we're creating excess clusters to satisfy unassigned claims, add that many
 	// to the runningCount. They'll get snatched up immediately, bringing the number
 	// of running clusters back down to runningCount once the pool reaches steady state.
-	runningCount := int(clp.Spec.RunningCount) + excessCount
-	cdList := append(cds.Assignable(), cds.Installing()...)
+	runningCount := int(clp.Spec.RunningCount) + extraRunning
```

Comment:
we are changing runningCount = spec.RunningCount + claims.Unassigned - cds.Installing - cds.Assignable - cds.Broken to runningCount = spec.RunningCount + claims.Unassigned
Questions still present about original implementation because number of excessCount was set before broken clusters were deleted but I digress
Consider spec.runningCount=0. Now if claims.Unassigned < cds.Unassigned(false), then we are good - but what happens when claims.Unassigned > cds.Unassigned(false)?
Wait - that wouldn't happen because we have already added those clusters at this point so they would be installing - right?

Reply:
Yeah, as noted above, the previous implementation was a bit sloppy, saved by the fact that we were ensuring the PowerState was set when we did the assignment.

Yes, by this point we have
- Deleted broken clusters. Though by definition none of those would have been running.
- Kicked off creation of new clusters, within `maxConcurrent`, to satisfy capacity and unclaimed clusters. Those also wouldn't be running yet.
- Assigned whatever claims we could. With this PR, those assignments consumed running clusters. Depending how many loops we've needed in non-steady state, that may or may not have reduced the number of running clusters below `runningCount`.

So at this point, we know we want the number of running clusters to be `runningCount` plus enough to accommodate whatever unassigned claims _remain_.

> Consider spec.runningCount=0. Now if claims.Unassigned < cds.Unassigned(false), then we are good - but what happens when claims.Unassigned > cds.Unassigned(false)?

I'm not sure if I've managed to answer this question with my ramblings above. Perhaps you can formulate it as a unit test case? Then we can either see if we already have that case covered, or create it if not.